cli/src/commands/commit.rs

```diff
@@ -106,6 +117,14 @@ new working-copy commit.
     if args.reset_author {
         commit_builder.set_author(commit_builder.committer().clone());
     }
+    if let Some((name, email)) = args.author.clone() {
+        let new_author = jj_lib::backend::Signature {
+            name,
+            email,
+            timestamp: commit_builder.committer().timestamp.clone(),
```

Comment:
you mean `author().timestamp`?

Reply:
This is intentional. using author's timestamp gives weird results:
```
-- STDOUT:              jj-cli::runner test_commit_command::test_commit_with_description_template ---

running 1 test
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Snapshot: commit_with_description_template-2
Source: cli/tests/test_commit_command.rs:244
────────────────────────────────────────────────────────────────────────────────
Expression: std::fs::read_to_string(test_env.env_root().join("editor")).unwrap()
────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬───────────────────────────────────────────────────────────────────
    0       │-JJ: Author: Another User <another.user@example.com> (2001-02-03 08:05:09)
          0 │+JJ: Author: Another User <another.user@example.com> (2001-02-03 08:05:08)
    1     1 │ JJ: Committer: Test User <test.user@example.com> (2001-02-03 08:05:09)
    2     2 │
    3     3 │ JJ: file2 | 1 +
    4     4 │ JJ: 1 file changed, 1 insertion(+), 0 deletions(-)
────────────┴───────────────────────────────────────────────────────────────────
To update snapshots run `cargo insta review`
Stopped on the first failure. Run `cargo insta test` to run all snapshots.
test test_commit_command::test_commit_with_description_template ... FAILED

failures:

failures:
    test_commit_command::test_commit_with_description_template

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 598 filtered out; finished in 0.89s
```
This would be inconsistent with other cases in that test.

Reply:
Well, if you don't pass `--author` here, you'll get the author timestamp of "2001-02-03 08:05:08", right? (Beware that the "file2" exists when the first commit is created (or split by `jj commit file1`.) Since `--author` shouldn't affect how author timestamp is updated, the tested behavior is wrong.

Reply:
Ah, that explains it.
Changed to author timestamp.